api/v1alpha1/kubevirtcluster_types.go

```diff
@@ -85,6 +88,16 @@ type SSHKeys struct {
 	DataSecretName *string `json:"dataSecretName,omitempty"`
 }
 
+// LoadBalancerServiceTemplate describes the template for the cluster load balancer service.
+type LoadBalancerServiceTemplate struct {
+	// +kubebuilder:pruning:PreserveUnknownFields
+	// +nullable
+	// Service metadata.
+	ObjectMeta metav1.ObjectMeta `json:"metadata,omitempty"`
+	// Service specification.
+	Spec corev1.ServiceSpec `json:"spec,omitempty" valid:"required"`
```

Comment:
Do you think it would make sense to introduce here a ServiceSpecTemplate? Otherwise it is pretty hard to understand which fields are considered. It seems to only be the `Type` right now. Right? API validation would let pass all kinds of fields which are finally ignored.

Reply:
yes that makes sense. originally, I thought we could expose the entire service spec to users (and it theory that could also work, we could set selector and ports in the template, too), but at the same time we might not want to complicate `KubevirtCluster` spec too much, so, for now I just exposed the service type, and service metadata, with potential to expose other service spec parts in future if needed. 

but agree, for now, we can just have our own ServiceSpecTemplate, to keep things explicit 👍

Reply:
also, note, that service spec is optional in KubevirtCluster object, and by default we fall back to the ClusterIP type as before. the main use case for now is being able to specify a different type (e.g. LoadBalancer) as well as setting labels and annotations for the service.